diff --git a/dist/esm/components.native.js b/dist/esm/components.native.js
index a080ce9..e812668 100644
--- a/dist/esm/components.native.js
+++ b/dist/esm/components.native.js
@@ -1,10 +1,12 @@
 import { jsx as _jsx, Fragment as _Fragment } from "react/jsx-runtime";
+import { useId } from "react";
 import { getPortal } from "./portalState.native.js";
 function NativePortal(param) {
   var {
     hostName = "root",
     children
   } = param;
+  var portalName = useId();
   var state = getPortal().state;
   if (state.type !== "teleport") return null;
   var {
@@ -12,6 +14,7 @@ function NativePortal(param) {
   } = globalThis.__tamagui_teleport;
   return /* @__PURE__ */_jsx(Portal, {
     hostName,
+    name: portalName,
     children
   });
 }
