diff --git a/dist/esm/ToastComposable.mjs b/dist/esm/ToastComposable.mjs
index 4828e8f..d563a48 100644
--- a/dist/esm/ToastComposable.mjs
+++ b/dist/esm/ToastComposable.mjs
@@ -710,7 +710,6 @@ const ToastItemInner = ToastItemFrame.styleable(function ToastItem(props, ref) {
   return /* @__PURE__ */jsx(ToastPositionWrapper, {
     ref,
     testID: rest.testID,
-    accessibilityLabel: rest.accessibilityLabel,
     ...dataAttributes,
     transition: isDragging || ctx.reducedMotion ? void 0 : removed ? "200ms" : "400ms",
     animateOnly: isWeb ? ["transform", "opacity", "height"] : ["transform", "opacity"],
diff --git a/dist/cjs/ToastComposable.cjs b/dist/cjs/ToastComposable.cjs
index 717cab0..3d8d38c 100644
--- a/dist/cjs/ToastComposable.cjs
+++ b/dist/cjs/ToastComposable.cjs
@@ -750,7 +750,6 @@ const ToastItemInner = ToastItemFrame.styleable(function ToastItem(props, ref) {
   return /* @__PURE__ */(0, import_jsx_runtime.jsx)(import_ToastItemFrame.ToastPositionWrapper, {
     ref,
     testID: rest.testID,
-    accessibilityLabel: rest.accessibilityLabel,
     ...dataAttributes,
     transition: isDragging || ctx.reducedMotion ? void 0 : removed ? "200ms" : "400ms",
     animateOnly: import_constants.isWeb ? ["transform", "opacity", "height"] : ["transform", "opacity"],