diff --git a/src/providers/animation-settings/provider.tsx b/src/providers/animation-settings/provider.tsx
index e3504b8..aad9532 100644
--- a/src/providers/animation-settings/provider.tsx
+++ b/src/providers/animation-settings/provider.tsx
@@ -1,4 +1,5 @@
 import React from 'react';
+import { Platform } from 'react-native';
 import { useReducedMotion } from 'react-native-reanimated';
 import { createContext } from '../../helpers/internal/utils';
 import type {
@@ -35,7 +36,7 @@ export const GlobalAnimationSettingsProviderComponent: React.FC<
 > = ({ animation, children }) => {
   const reducedMotion = useReducedMotion();
   const globalIsAllAnimationsDisabled =
-    animation === 'disable-all' || reducedMotion;
+    animation === 'disable-all' || (Platform.OS !== 'web' && reducedMotion); 
 
   return (
     <GlobalAnimationSettingsProvider value={{ globalIsAllAnimationsDisabled }}>
diff --git a/lib/module/providers/animation-settings/provider.js b/lib/module/providers/animation-settings/provider.js
index 7a8c9b2..9181c90 100644
--- a/lib/module/providers/animation-settings/provider.js
+++ b/lib/module/providers/animation-settings/provider.js
@@ -1,6 +1,7 @@
 "use strict";
 
 import React from 'react';
+import { Platform } from 'react-native';
 import { useReducedMotion } from 'react-native-reanimated';
 import { createContext } from "../../helpers/internal/utils/index.js";
 import { jsx as _jsx } from "react/jsx-runtime";
@@ -31,7 +32,7 @@ export const GlobalAnimationSettingsProviderComponent = ({
   children
 }) => {
   const reducedMotion = useReducedMotion();
-  const globalIsAllAnimationsDisabled = animation === 'disable-all' || reducedMotion;
+  const globalIsAllAnimationsDisabled = animation === 'disable-all' || Platform.OS !== 'web' && reducedMotion;
   return /*#__PURE__*/_jsx(GlobalAnimationSettingsProvider, {
     value: {
       globalIsAllAnimationsDisabled
